src/item_location.cpp

```diff
@@ -398,22 +392,14 @@ class item_location::impl::item_on_person : public item_location::impl
             }
 
             int mv = 0;
-
             item *obj = target();
-            item temp;
-            if( target()->count_by_charges() ) {
-                temp = target()->split( qty );
-                if( !temp.is_null() ) {
-                    obj = &temp;
-                }
-            }
-
-            item &target_ref = *target();
-            if( who->is_wielding( target_ref ) ) {
-                mv = who->item_handling_cost( *obj, false, 0 );
+            int handled_qty = obj->count_by_charges() ? ( qty <= 0 ||
+                              qty >= obj->charges ) ? obj->charges : qty : -1;
```

Comment:
Doesn't this accomplish the same thing?
```suggestion
            int handled_qty = std::clamp( qty, -1, obj->count() );
```

Reply:
Yes, and another look into item::volume() even makes me think that qty can be directly passed to it.....